vmaas-go/base/utils/gin.go

```diff
@@ -105,6 +106,25 @@ func respStatusError(c *gin.Context, code int, err error) {
 	})
 }
 
+func processBadRequestErrMessage(err error) error {
+	errMessage := err.Error()
+	if strings.HasPrefix(errMessage, "parsing time") {
+		parts := strings.Split(errMessage, `"`)
+		if len(parts) < 2 {
+			return errors.New("Wrong date format (not ISO format with timezone)")
+		}
+		return errors.New("Wrong date format (not ISO format with timezone): " + parts[1])
```

Comment:
what's the purpose of splitting the error message? When it has only 1 `"` then the error will be `Wrong date format (not ISO format with timezone` and when there are more `"` then the second part is wrapped by the new message and other parts are discarded?
If the point is to only wrap the current error, you can use `errors.Wrap` from `github.com/pkg/errors` which is already used in this module or https://pkg.go.dev/errors#Join

Reply:
The purpose is to get the original timestamp being parsed, the error message without one is just a fallback.